core/imageutil/imaging/demo/main.go

```diff
@@ -0,0 +1,150 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"image"
+	"io"
+	"io/fs"
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	_ "image/gif"
+	_ "image/jpeg"
+	_ "image/png"
+
+	_ "golang.org/x/image/bmp"
+	_ "golang.org/x/image/ccitt"
+	_ "golang.org/x/image/riff"
+	_ "golang.org/x/image/tiff"
+	_ "golang.org/x/image/vector"
+	_ "golang.org/x/image/vp8"
+	_ "golang.org/x/image/vp8l"
+	_ "golang.org/x/image/webp"
+
+	"github.com/0chain/gosdk/core/imageutil/imaging"
+)
+
+type img struct {
+	// location to download image
+	location string
+	// location type can be either remote or local
+	locationType string
+	// expected width of thumbnail
+	width int
+	// expected height of thumbnail
+	height int
+	// ResampleFilter
+	resample imaging.ResampleFilter
+}
+
+var (
+	permissions = 0644
+	images      = []img{
+		{
+			location:     "https://go.dev/blog/go-brand/Go-Logo/JPG/Go-Logo_Aqua.jpg",
+			locationType: "remote",
+			width:        100,
+			height:       100,
+		},
+		{
+			location:     "https://go.dev/blog/go-brand/Go-Logo/PNG/Go-Logo_Yellow.png",
+			locationType: "remote",
+			width:        200,
+			height:       200,
+		},
+		{
+			location:     filepath.Join("..", "..", "resources", "input.gif"),
+			locationType: "local",
+			width:        100,
+			height:       200,
+		},
+	}
+)
+
+func main() {
```

Comment:
can it be converted to test?

Reply:
test file is present. this file is just for demo purpose.